music_assistant/providers/ard_audiothek/__init__.py

```diff
@@ -0,0 +1,808 @@
+                duration = x["item"]["duration"]
+                if duration is None:
+                    continue
+                progress = x["progress"]
+                time_limit_reached = (progress / duration) * 100 > time_limit
+                new_progress[core_id] = (time_limit_reached, progress)
+            self.remote_progress = new_progress
+
+    def _get_progress(self, episode_id: str) -> tuple[bool, int]:
+        if episode_id in self.remote_progress:
+            return self.remote_progress[episode_id][0], int(
+                self.remote_progress[episode_id][1] * 1000
+            )
+        return False, 0
+
+    async def get_resume_position(self, item_id: str, media_type: MediaType) -> tuple[bool, int]:
+        """Return: finished, position_ms."""
+        assert media_type == MediaType.PODCAST_EPISODE
+        await self._update_progress()
+
+        return self._get_progress(item_id)
+
+    async def on_played(
+        self,
+        media_type: MediaType,
+        prov_item_id: str,
+        fully_played: bool,
+        position: int,
+        media_item: MediaItemType,
+        is_playing: bool = False,
+    ) -> None:
+        """Update progress."""
+        if not self.user_id:
+            return
+        if media_item is None or not isinstance(media_item, PodcastEpisode):
+            return
+        if media_type != MediaType.PODCAST_EPISODE:
+            return
+        async with await self.get_client() as session:
+            update_history_entry.variable_values = {"itemId": prov_item_id, "progress": position}
+            await session.execute(
+                update_history_entry,
+            )
+
+    @property
+    def is_streaming_provider(self) -> bool:
+        """Search and lookup always search remote."""
+        # For streaming providers return True here but for local file based providers return False.
+        return True
+
+    async def search(
+        self,
+        search_query: str,
+        media_types: list[MediaType],
+        limit: int = 5,
+    ) -> SearchResults:
+        """Perform search on musicprovider.
+
+        :param search_query: Search query.
+        :param media_types: A list of media_types to include.
+        :param limit: Number of items to return in the search (per type).
+        """
+        async with await self.get_client() as session:
+            search_shows_query.variable_values = {"query": search_query, "limit": limit}
+            search_shows = (await session.execute(search_shows_query))["search"]["shows"]["nodes"]
+
+        podcasts = []
+        for element in search_shows:
+            podcasts += [
+                _parse_podcast(
+                    self.domain,
+                    self.lookup_key,
+                    self.instance_id,
+                    element,
+                    element["coreId"],
+                )
+            ]
+        async with await self.get_client() as session:
+            search_radios_query.variable_values = {
```

Comment:
Look at line 379 as well. Doesn't this mean that two API calls are made for every search even if you are just looking for one type?